mitodl/ocw-studio
# Get site/file url prefix from site config

#### Pre-Flight checklist



#### What are the relevant tickets?
- Partially addresses #336 

#### What's this PR do?
- Adds `base-url` to site configs and makes the field mandatory
- Uploads files with the `base-url` value as a prefix in the filepath.

#### How should this be manually tested?
- Run `docker-compose run web python manage.py override_site_config` to add `base-url` to the default site configs.
- If you have other site configs besides the defaults, you should add `base-url: <any-value>` to each.
- Create a new site and add a resource, uploading a file.  Check the filepath of the uploaded file, it should start with the `base-url` value specified in the site config and then the site name: `<base_url>/<site_name>`.


#### Any background context you want to provide?
I decided against a management command for updating existing filepaths, since there's no real data yet on production.  Let me know if you feel otherwise.
